qutip/solver/sode/_sode.pyx

```diff
@@ -0,0 +1,510 @@
+
+        system.set_state(t, state)
+
+        imul_dense(out, 0.)
+        iadd_dense(out, state, 1)
+        iadd_dense(out, system.a(), dt * (1-alpha))
+
+        imul_dense(euler, 0.)
+        iadd_dense(euler, state, 1)
+        iadd_dense(euler, system.a(), dt)
+
+        for i in range(num_ops):
+            iadd_dense(euler, system.bi(i), dW[0, i])
+            iadd_dense(out, system.bi(i), dW[0, i] * eta)
+            iadd_dense(out, system.Libj(i, i), dt * (alpha-1) * 0.5)
+
+        system.set_state(t+dt, euler)
+        for i in range(num_ops):
+            iadd_dense(out, system.bi(i), dW[0, i] * (1-eta))
+
+        if alpha:
+            iadd_dense(out, system.a(), dt*alpha)
+            for i in range(num_ops):
+                iadd_dense(out, system.Libj(i, i), -dt * alpha * 0.5)
+
+        return out
+
+
+cdef class Taylor15(Milstein):
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    cdef step(self, double t, Dense state, double dt, double[:, :] dW, Dense out):
+        """
+        Chapter 10.4 Eq. (4.6),
+        Numerical Solution of Stochastic Differential Equations
+        By Peter E. Kloeden, Eckhard Platen
+        """
+        cdef _StochasticSystem system = self.system
+        system.set_state(t, state)
+        cdef int i, j, k, num_ops = system.num_collapse
+        cdef double[:] dz, dw
+
+        num_ops = system.num_collapse
+        dw = dW[0, :]
+        dz = 0.5 * (dW[0, :] + dW[1, :] / np.sqrt(3)) * dt
+
+        imul_dense(out, 0.)
+        iadd_dense(out, state, 1)
+        iadd_dense(out, system.a(), dt)
+        iadd_dense(out, system.L0a(), 0.5 * dt * dt)
+
+        for i in range(num_ops):
+            iadd_dense(out, system.bi(i), dw[i])
+            iadd_dense(out, system.Libj(i, i), 0.5 * (dw[i] * dw[i] - dt))
+            iadd_dense(out, system.Lia(i), dz[i])
+            iadd_dense(out, system.L0bi(i), dw[i] * dt - dz[i])
+            iadd_dense(out, system.LiLjbk(i, i, i),
+                             0.5 * ((1/3.) * dw[i] * dw[i] - dt) * dw[i])
+
+            for j in range(i+1, num_ops):
+                iadd_dense(out, system.Libj(i, j), dw[i] * dw[j])
+                iadd_dense(out, system.LiLjbk(i, j, j), 0.5 * (dw[j] * dw[j] -dt) * dw[i])
+                iadd_dense(out, system.LiLjbk(i, i, j), 0.5 * (dw[i] * dw[i] -dt) * dw[j])
+                for k in range(j+1, num_ops):
+                    iadd_dense(out, system.LiLjbk(i, j, k), dw[i]*dw[j]*dw[k])
+
+        return out
+
+
+cdef class Milstein_imp:
+    cdef _StochasticSystem system
+    cdef bint use_inv
+    cdef QobjEvo implicit
+    cdef Data inv
+    cdef double prev_dt
+    cdef dict imp_opt
+
+    def __init__(self, _StochasticSystem system, imp_method=None, imp_options={}):
+        self.system = system
+        self.prev_dt = 0
+        if imp_method == "inv":
+            if not self.system.L.isconstant:
+                raise TypeError("")
+            self.use_inv = True
```

Comment:
```suggestion
            self.use_inv = True
            self.imp_opt = {}
```
Probably makes sense to initialize `imp_opt` here explicitly too even if it isn't used.